packages/@react-types/shared/src/dnd.d.ts

```diff
@@ -127,12 +163,21 @@ export interface DropTargetDelegate {
   getDropTargetFromPoint(x: number, y: number, isValidDropTarget: (target: DropTarget) => boolean): DropTarget | null
 }
 
+// TODO: naming?
+export interface DropOperationEvent {
+  target: DropTarget,
+  types: DragTypes,
+  allowedOperations: DropOperation[],
+  isInternalDrop: boolean,
+  draggingKeys: Set<Key>
+}
+
 export interface DroppableCollectionProps {
   /**
    * A function returning the drop operation to be performed when items matching the given types are dropped
    * on the drop target.
    */
-  getDropOperation?: (target: DropTarget, types: DragTypes, allowedOperations: DropOperation[]) => DropOperation,
+  getDropOperation?: (e: DropOperationEvent) => DropOperation,
```

Comment:
If we don't expose `isInternalDrop`/`draggingKeys` to the prop, and keep that internal, then we could keep the API the same here.